### PR TITLE
feat(networkconnectivity): Add samples for creating a service-connection-policy

### PR DESCRIPTION
**Pull request by skratochvil**
## Description

I'm creating this as a draft PR because I haven't been able to run the tests at https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment. I was able to try `terraform apply` and verify that it created the right resources in my test project. 

Fixes #302014019

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**
I have not been able to complete the tests below. After following the steps in the guide, `make -s docker_test_sample SAMPLE=networkconnectivity_create_service_connection_policy` ends with
```
TestSamples 2023-09-26T18:48:32Z command.go:185: ["ci-tf-samples-0-531d","ci-tf-samples-1-9fe6","ci-tf-samples-2-fda6","ci-tf-samples-3-8a5b"]
--- PASS: TestSamples (1.74s)
testing: warning: no tests to run
PASS
ok      github.com/terraform-google-modules/terraform-docs-samples/test/integration     1.759s
```
- [ ] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/vpc/docs/configure-service-connection-policies#create-policy

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [X] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

Added sample under the existing network_connectivity directory.

- [X] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by msampathkumar** [comment]
/gcbrun

invoking CICD

File: network_connectivity/service_connectivity_automation/service_connection_policies/main.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+resource "google_compute_subnetwork" "default" {
+  name          = "consumer-subnet"
+  ip_cidr_range = "10.0.0.0/16"
+  region        = "us-central1"
+  network       = google_compute_network.default.id
+}
+
+# Create a service connection policy
+resource "google_network_connectivity_service_connection_policy" "default" {
+  name          = "my-service-connection-policy"
```
Comment (msampathkumar) [inline network_connectivity/service_connectivity_automation/service_connection_policies/main.tf:34]:
```suggestion
  name          = "service-connection-policy"
```

Suggestion to drop `my-` prefix for resource name. Rest of the resource don't have have this `my-`

**Comment by msampathkumar** [comment]
/gcbrun

Triggering the Integrations CICD test

**Review by gericdong — APPROVED** [review]
Integration failing on an irrelevant test.
LGTM.

**Pull request merged**